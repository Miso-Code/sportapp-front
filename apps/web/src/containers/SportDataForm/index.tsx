import SelectController from '@/components/Inputs/SelectController'
import TextFieldController from '@/components/Inputs/TexFieldController'
import { yupResolver } from '@hookform/resolvers/yup'
import LoadingButton from '@mui/lab/LoadingButton'
import {
	Button,
	Card,
	CardContent,
	Divider,
	List,
	ListItem,
	TextField,
	Typography
} from '@mui/material'
import { useSportStore } from '@sportapp/stores/src/sport'
import { Props } from 'containers/SportDataForm/interfaces'
import { useCallback, useEffect, useMemo } from 'react'
import { useFieldArray, useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import './_index.scss'
import {
	FormDataBase,
	FormDataRequired,
	schemaBase,
	schemaRequired
} from './utils/schema'

export default function SportDataForm({
	className = '',
	inputsDisabled,
	isDisabled,
	defaultValues,
	customSubmit,
	isRequired,
	handleCustomSubmit
}: Props) {
	const { t } = useTranslation()
	const { getSports } = useSportStore()
	const { sports } = useSportStore()
	const { handleSubmit, control, register } = useForm({
		resolver: yupResolver(isRequired ? schemaRequired : schemaBase),
		defaultValues: {
			...defaultValues
		},
		mode: 'onChange'
	})
	const { fields, append, remove } = useFieldArray({
		control,
		name: 'limitations'
	})

	const onSubmit = (data: FormDataRequired | FormDataBase) => {
		handleCustomSubmit(data)
	}

	const handleGetSports = useCallback(async () => {
		await getSports()
	}, [getSports])

	const getFormatSport = useMemo(() => {
		if (Array.isArray(sports))
			return sports.map((sport) => ({
				label: sport.name,
				value: sport.sport_id
			}))
		else return []
	}, [sports])

	useEffect(() => {
		handleGetSports()
	}, [handleGetSports])

	return (
		<div className={`personal-data-form ${className}`}>
			<Typography className='personal-data-form-title' variant='h3'>
				{t('sportDataForm.title')}
			</Typography>
			<form
				className='personal-data-form-container'
				onSubmit={handleSubmit(onSubmit)}>
				<SelectController
					control={control}
					selectProps={{ fullWidth: true }}
					label={t('form.favouriteSport')}
					name='favouriteSportId'
					formControlProps={{ disabled: inputsDisabled }}
					options={getFormatSport}
				/>

				<SelectController
					control={control}
					selectProps={{ fullWidth: true }}
					label={t('form.trainingObjective')}
					name='trainingObjective'
					formControlProps={{ disabled: inputsDisabled }}
					options={[
						{
							label: 'form.trainingObjectiveValue.build_muscle_mass',
							value: 'build_muscle_mass'
						},
						{
							label: 'form.trainingObjectiveValue.lose_weight',
							value: 'lose_weight'
						},
						{
							label: 'form.trainingObjectiveValue.tone_up',
							value: 'tone_up'
						},
						{
							label: 'form.trainingObjectiveValue.maintain_fitness',
							value: 'maintain_fitness'
						}
					]}
				/>

				<SelectController
					control={control}
					selectProps={{ fullWidth: true }}
					label={t('form.trainingFrequency')}
					name='trainingFrequency'
					formControlProps={{ disabled: inputsDisabled }}
					options={[
						{
							label: 'form.trainingFrequencyValues.DAILY',
							value: 'daily'
						},
						{
							label: 'form.trainingFrequencyValues.EVERY_OTHER_DAY',
							value: 'every_other_day'
						},
						{
							label: 'form.trainingFrequencyValues.WEEKLY',
							value: 'weekly'
						},
						{
							label: 'form.trainingFrequencyValues.MONTHLY',
							value: 'monthly'
						}
					]}
				/>

				{/* <SelectController
					control={control}
					selectProps={{ fullWidth: true, multiple: true }}
					label={t('form.limitations')}
					name='limitations'
					formControlProps={{ disabled: inputsDisabled }}
					options={[
						{
							label: t('form.trainingFrequencyValues.DAILY'),
							value: 'daily'
						},
						{
							label: t(
								'form.trainingFrequencyValues.EVERY_OTHER_DAY'
							),
							value: 'every_other_day'
						},
						{
							label: t('form.trainingFrequencyValues.WEEKLY'),
							value: 'weekly'
						},
						{
							label: t('form.trainingFrequencyValues.MONTHLY'),
							value: 'monthly'
						}
					]}
				/> */}

				<Card className='w-full'>
					<CardContent>
						<Typography
							color={inputsDisabled ? 'GrayText' : 'InfoText'}
							variant='subtitle2'
							className='mb-2'>
							{t('form.limitations')}
						</Typography>
						<List>
							{fields.map((field, index) => (
								<>
									<Divider component='li' className='mb-4' />
									<Typography
										variant='caption'
										color={
											inputsDisabled
												? 'GrayText'
												: 'InfoText'
										}
										className='mb-2'>
										{`${t('form.limitation').toLowerCase()} #${index + 1}:`}
									</Typography>
									<ListItem className='pl-4 flex-col'>
										<div
											key={field.id}
											className='flex flex-col gap-4 mb-2 w-full'>
											<TextField
												{...register(
													`limitations.${index}.name`
												)}
												fullWidth
												disabled={inputsDisabled}
												label={t(
													'form.limitationsLabels.name'
												)}
												name={`limitations.${index}.name`}
											/>
											<TextField
												{...register(
													`limitations.${index}.description`
												)}
												fullWidth
												disabled={inputsDisabled}
												label={t(
													'form.limitationsLabels.description'
												)}
												name={`limitations.${index}.description`}
											/>
										</div>
										<Button
											color='error'
											type='button'
											disabled={inputsDisabled}
											onClick={() => remove(index)}>
											{`${t('form.limitationsLabels.remove')} #${index + 1}`}
										</Button>
									</ListItem>
								</>
							))}
						</List>

						<Button
							type='button'
							color='success'
							disabled={inputsDisabled}
							onClick={() =>
								append({ description: '', name: '' })
							}>
							{t('form.limitationsLabels.add')}
						</Button>
					</CardContent>
				</Card>

				<TextFieldController
					control={control}
					fullWidth
					disabled={inputsDisabled}
					label={t('form.availableTrainingHoursPerWeek')}
					name='availableTrainingHoursPerWeek'
					type='number'
				/>

				<TextFieldController
					control={control}
					fullWidth
					disabled={inputsDisabled}
					label={t('form.weight')}
					name='weight'
					type='number'
				/>

				<TextFieldController
					control={control}
					fullWidth
					disabled={inputsDisabled}
					label={t('form.height')}
					name='height'
					type='number'
				/>

				<TextFieldController
					control={control}
					fullWidth
					disabled
					label={t('form.imc')}
					name='imc'
					type='number'
				/>

				{!!customSubmit || (
					<LoadingButton
						disabled={isDisabled}
						loading={isDisabled}
						fullWidth
						size='large'
						type='submit'
						variant='contained'>
						{t('sportDataForm.save')}
					</LoadingButton>
				)}
			</form>
			{customSubmit}
		</div>
	)
}
