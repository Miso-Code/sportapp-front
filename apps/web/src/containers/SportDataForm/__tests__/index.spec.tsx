import SportDataForm from '..'
import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider, createTheme } from '@mui/material'

describe('SportDataForm', () => {
	const mockTheme = createTheme({
		palette: {
			primary: {
				main: '#ffffff',
				light: '#ffffff'
			},
			text: {
				disabled: '#ffffff'
			},
			action: {
				disabledBackground: '#ffffff'
			}
			// Add more properties as needed
		}
	})
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(
			<ThemeProvider theme={mockTheme}>
				<SportDataForm handleCustomSubmit={jest.fn()} />
			</ThemeProvider>
		)
	})

	afterEach(() => {
		wrapper.unmount()
	})

	it('should render the component', () => {
		expect(wrapper.container).toMatchSnapshot()
	})
})
