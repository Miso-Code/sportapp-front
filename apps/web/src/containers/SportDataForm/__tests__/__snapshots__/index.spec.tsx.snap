// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SportDataForm should render the component 1`] = `
<div>
  <div
    class="personal-data-form "
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 personal-data-form-title css-gepadz-MuiTypography-root"
    >
      sportDataForm.title
    </h3>
    <form
      class="personal-data-form-container"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-16zti8r-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="favouriteSportId-select-label"
        >
          form.favouriteSport
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl  css-u8zffx-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-controls=":r0:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="favouriteSportId-select-label mui-component-select-favouriteSportId"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-e890gg-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="mui-component-select-favouriteSportId"
            role="combobox"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            id="favouriteSportId"
            name="favouriteSportId"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                form.favouriteSport
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-16ojagh-MuiFormHelperText-root"
        />
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-16zti8r-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="trainingObjective-select-label"
        >
          form.trainingObjective
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl  css-u8zffx-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-controls=":r1:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="trainingObjective-select-label mui-component-select-trainingObjective"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-e890gg-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="mui-component-select-trainingObjective"
            role="combobox"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            id="trainingObjective"
            name="trainingObjective"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                form.trainingObjective
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-16ojagh-MuiFormHelperText-root"
        />
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed px-0 css-1ss2ei0-MuiContainer-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 mb-2 css-1ptzdzb-MuiTypography-root"
        >
          form.daysOfTheWeek
        </h6>
        <div
          class="css-iglq0z"
        >
          <div
            aria-label="Days of the week"
            class="MuiToggleButtonGroup-root css-f8i28h-MuiToggleButtonGroup-root"
            role="group"
          >
            <button
              aria-label="sunday"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-firstButton css-1rijrxi-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="sunday"
            >
              form.daysOfTheWeekValues.SUNDAY
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="monday"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-middleButton css-1rijrxi-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="monday"
            >
              form.daysOfTheWeekValues.MONDAY
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="tuesday"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-middleButton css-1rijrxi-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="tuesday"
            >
              form.daysOfTheWeekValues.TUESDAY
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="wednesday"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-middleButton css-1rijrxi-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="wednesday"
            >
              form.daysOfTheWeekValues.WEDNESDAY
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="thursday"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-middleButton css-1rijrxi-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="thursday"
            >
              form.daysOfTheWeekValues.THURSDAY
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="friday"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-middleButton css-1rijrxi-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="friday"
            >
              form.daysOfTheWeekValues.FRIDAY
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="saturday"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-lastButton css-1rijrxi-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="saturday"
            >
              form.daysOfTheWeekValues.SATURDAY
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-n5zogd-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-16zti8r-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="preferedTrainingStartTime"
          id="preferedTrainingStartTime-label"
        >
          form.preferedTrainingStartTime
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-53lqbx-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-v8xe14-MuiInputBase-input-MuiOutlinedInput-input"
            id="preferedTrainingStartTime"
            inputmode="text"
            name="preferedTrainingStartTime"
            placeholder="hh:mm aa"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose time"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ClockIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
                <path
                  d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                form.preferedTrainingStartTime
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-16zti8r-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r4:"
          id=":r4:-label"
        >
          form.availableTrainingHoursPerDay
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-53lqbx-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-v8xe14-MuiInputBase-input-MuiOutlinedInput-input"
            id="availableTrainingHoursPerDay"
            name="availableTrainingHoursPerDay"
            type="number"
            value="0"
          />
          h
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                form.availableTrainingHoursPerDay
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-16zti8r-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r5:"
          id=":r5:-label"
        >
          form.weight
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-53lqbx-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-v8xe14-MuiInputBase-input-MuiOutlinedInput-input"
            id="weight"
            name="weight"
            type="number"
            value="0"
          />
          kg
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                form.weight
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-16zti8r-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r6:"
          id=":r6:-label"
        >
          form.height
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-53lqbx-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-v8xe14-MuiInputBase-input-MuiOutlinedInput-input"
            id="height"
            name="height"
            type="number"
            value="0"
          />
          m
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                form.height
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-16zti8r-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r7:"
          id=":r7:-label"
        >
          form.imc
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-53lqbx-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-v8xe14-MuiInputBase-input-MuiOutlinedInput-input"
            disabled=""
            id="imc"
            name="imc"
            type="number"
            value="0"
          />
          %
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                form.imc
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root w-full css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 mb-2 css-1ptzdzb-MuiTypography-root"
          >
            form.limitations
          </h6>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSuccess css-1lg55su-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            form.limitationsLabels.add
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth css-1k1a1sp-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        id=":r8:"
        tabindex="0"
        type="submit"
      >
        sportDataForm.save
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;
